addons/sale_stock/models/sale_order_line.py

```diff
@@ -388,3 +388,6 @@ def _get_product_catalog_lines_data(self, **kwargs):
             )
         )
         return res
+
+    def has_valued_move_ids(self):
+        return any(move.state not in ('cancel', 'draft') for move in self.move_ids)
```

Comment:
```suggestion
    def has_valued_move_ids(self):
        res = super().has_valued_move_ids()
        return res and any(move.state not in ('cancel', 'draft') for move in self.move_ids)
```
also make sure it doesn't conflict with `repair/has_valued_move_ids`

Reply:
How can it conflict when the method is moved to the class that adds the `move_ids` field? It doesn't make sense for any other method to be the `super` call.